sunpy/map/mapbase.py

```diff
@@ -1615,15 +1616,23 @@ def rotate(self, angle: u.deg = None, rmatrix=None, order=4, scale=1.0,
             rmatrix = np.array([[c, -s],
                                 [s, c]])
 
+        # The data will be rotated by the inverse of the rotation matrix
+        inv_rmatrix = np.linalg.inv(rmatrix)
```

Comment:
I opted not to rename it, partially because `rmatrix` is the name of the input parameter.  Instead, I added comments in two places to help future developers.  I'll point out that the potential confusion here is not as bad as the fact that `GenericMap.rotation_matrix` doesn't always return a pure rotation matrix. =)